### PR TITLE
core: don't use context.ToDo everywhere

### PR DESCRIPTION
**Pull request by subhamkrai**
core: don't use context.ToDo everywhere

using context.ToDo always is not a good practice.
So, now having a central place of context which
all methods will use.

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

**Comment by travisn** [comment]
Do you see a difference in the behavior of killing a running command with Control-C with this context change? I'm assuming it will cancel the request sooner.

**Comment by subhamkrai** [comment]
> Do you see a difference in the behavior of killing a running command with Control-C with this context change? I'm assuming it will cancel the request sooner.

well for me, it is fast with or without these changes I am hardly able to find a diff but when I used `time ` command I see diff

without these changes
```yaml
real	0m0.589s
user	0m0.028s
sys	0m0.014s
```

with these changes
```yaml
real	0m0.387s
user	0m0.052s
sys	0m0.022s
```

**Comment by subhamkrai** [comment]
will rebase once logging pr is merged

File: cmd/commands/root.go [hunk 1]
```diff
@@ -84,5 +86,7 @@ func GetContext() *k8sutil.Context {
 		logging.Fatal(err)
 	}
 
+	context.Context = ctx.TODO()
```
Comment (travisn) [inline cmd/commands/root.go:89]: I thought the intention was to not use `ctx.TODO()`.

Comment (subhamkrai) [reply cmd/commands/root.go]: now, using cobra root command context. Similar to multus validation in rook

**Review by travisn — APPROVED** [review]

**Pull request merged**